tests/game/test_buildings.py

```diff
@@ -218,3 +220,393 @@ def test_tree_wildanimalfood_production(s, p):
 
 	# after producing 20 units of food inventory should be empty
 	assert inv[RES.WILDANIMALFOOD] == 0
+
+
+@game_test()
+def test_weaver_production_chain(s, p):
+	"""
+	Lamb wool is generated by sheep at a pasture to be later transferred to a weaver
+	for textile production
+	"""
+	settlement, island = settle(s)
+	
+	assert Build(BUILDINGS.FARM, 30, 30, island, settlement=settlement)(p)
+	assert Build(BUILDINGS.PASTURE, 26, 30, island, settlement=settlement)(p)
+	
+	weaver = Build(BUILDINGS.WEAVER, 30, 26, island, settlement=settlement)(p)
+	
+	assert weaver
+	
+	assert weaver.get_component(StorageComponent).inventory[RES.WOOL] == 0
+	assert weaver.get_component(StorageComponent).inventory[RES.TEXTILE] == 0
+	s.run(seconds=60) # 30s pasture, 30s weaver
+	assert weaver.get_component(StorageComponent).inventory[RES.TEXTILE]
+
+
+@game_test()
+def test_pavilion_production(s, p):
+	"""
+	Check whether the pavilion produces faith
+	"""
+	settlement, island = settle(s)
+	pavilion = Build(BUILDINGS.PAVILION, 30, 30, island, settlement=settlement)(p)
+
+	assert pavilion
+	assert pavilion.get_component(StorageComponent).inventory[RES.FAITH] == 0
+	s.run(seconds=30)
+	assert pavilion.get_component(StorageComponent).inventory[RES.FAITH]
+
+
+@game_test()
+def test_farm_production(s, p):
```

Comment:
Unfortunately, `_build_farm` for now would create 4 fields of the same type. I will see how to modify the function in such a way that more fields of different types can be placed next to one or more farms.